File: src/main/java/com/playground/springbootscala/SpringBootScalaApplication.java
package com.playground.springbootscala;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;

/**
 * We tried various Scala implementation of this class, none of them can survive
 * {@code mvn clean test} <br/>
 * The issue is spring-boot plugin can not find main class. This might related to the fact that
 * Scala companion object is not really a static method replacer
 * @author CX无敌
 */
@SpringBootApplication
public class SpringBootScalaApplication {
  public static void main(String[] args) {
    SpringApplication.run(SpringBootScalaApplication.class, args);
  }
}

File: src/main/scala/com/playground/springbootscala/SpringBootScalaApplication.scala
package com.playground.springbootscala

import org.springframework.boot.SpringApplication
import org.springframework.boot.autoconfigure.SpringBootApplication

@SpringBootApplication
class SpringBootScalaApplication {// must declare the class, object is a final class which prevent running .run() function
  // the worst disadvantage of using Scala is intellij only shows Scala types in auto-complete by default,
  // to show Java type, either press ctrl+space again, or finish the typing
}

object SpringBootScalaApplication {
  def main(args: Array[String]): Unit = {
    SpringApplication.run(classOf[SpringBootScalaApplication], args:_*) //don't use this.type, we want the class not the object
  }
}



